src/InternalUtils.vala

```diff
@@ -122,19 +122,18 @@ namespace Gala {
 
                     rects = {topleft, topright, bottomleft, bottomright};
 
-                    // add plugin's requested areas
-                    if (area == InputArea.FULLSCREEN || area == InputArea.DEFAULT) {
-                        foreach (var rect in PluginManager.get_default ().regions) {
-                            rects += rect;
-                        }
-                    }
                     break;
                 case InputArea.NONE:
                 default:
                     unowned Meta.X11Display x11display = display.get_x11_display ();
                     x11display.clear_stage_input_region ();
                     return;
             }
+            // Must be InputArea.FULLSCREEN or InputArea.DEFAULT if reach here
+            // add plugin's requested areas
+            foreach (var rect in PluginManager.get_default ().regions) {
```

Comment:
While it's true that `if (area == InputArea.FULLSCREEN || area == InputArea.DEFAULT)` was redundant, this `foreach` was called only when `area == InputArea.DEFAULT`, wasn't it?

I think that calling it here would change the previous behaviour where this wasn't performed when `area == InputArea.FULLSCREEN`.

Having said that, I don't know it that could cause bugs.

Reply:
Yes, I am not totally sure what the intended behaviour actually was - what the if clause implies or what the switch clause implies. :disappointed: I may be safer not to add plugin areas to FULLSCREEN WINDOWS I guess.

Reply:
Yes, it is difficult to tell who is lying, the if or the switch xD But just in case, I'd try to keep the behaviour unchanged, just in case.

Reply:
Done